File: frontend/src/shared/models/sub-flow/sub-flow-create-dto.ts
import type { FlowSearchAreaCreateDto } from "@/shared/models/flow-search-area/flow-search-area-create-dto";

export class SubFlowCreateDto {
  // id: number = -1;
  name: string = "";
  orderNumber: number = -1;

  // flowSteps: FlowStepCreateDto[] = [];
  flowSearchAreas: FlowSearchAreaCreateDto[] = [];
}

File: backend/Core/Models/Dtos/Database/SubFlow/SubFlowCreateDto.cs
using System.Collections.ObjectModel;

namespace Core.Models.Dtos
{
    public class SubFlowCreateDto
    {
        public string Name { get; set; } = string.Empty;
        public int OrderNumber { get; set; }

        public virtual IEnumerable<FlowStepCreateDto> FlowSteps { get; set; } = new Collection<FlowStepCreateDto>();
        public virtual IEnumerable<FlowSearchAreaCreateDto> FlowSearchAreas { get; set; } = new Collection<FlowSearchAreaCreateDto>();
    }
}
